File: test/05_test.dart
import 'dart:io';
import 'package:test/test.dart';
import 'package:advent_of_code_2020/05.dart';

void main() {
  String contents = new File('./data/05.txt').readAsStringSync();
  
  group('A', () {
    test('Example 1', () {
      expect(A05('FBFBBFFRLR\nBFFFBBFRRR\nFFFBBBFRRR\nBBFFBBFRLL'), equals(820));
    });
    test('Solution', () {
      expect(A05(contents), equals(883));
    });
  });

  group('B', () {
    test('Solution', () {
      expect(B05(contents), equals(532));
    });
  });
}

File: test/05.test.js
const fs = require('fs');

const { A05, B05 } = require('../lib/05');
const input = fs.readFileSync('./data/05.txt', 'utf-8');

describe('05', () => {
	describe('A', () => {
		it('Example 1', () => {
			expect(A05(`FBFBBFFRLR\nBFFFBBFRRR\nFFFBBBFRRR\nBBFFBBFRLL`)).toBe(820);
		});

		it('Solution', () => {
			expect(A05(input)).toBe(883);
		});
	});

	describe('B', () => {
		it('Solution', () => {
			expect(B05(input)).toBe(532);
		});
	});
});
